test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelperSampleTest.cs

```diff
@@ -18,44 +23,164 @@ public TagHelperSampleTest(MvcTestFixture<TagHelperSample.Web.Startup> fixture)
+            var resourceAssembly = typeof(TagHelperSampleTest).GetTypeInfo().Assembly;
+            var expectedContent = await ResourceFile.ReadResourceAsync(resourceAssembly, outputFile, sourceFile: false);
+
+            // Act
+            var response = await Client.GetAsync("http://localhost/");
+            var responseContent = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(expectedMediaType, response.Content.Headers.ContentType);
+
+#if GENERATE_BASELINES
+            ResourceFile.UpdateFile(resourceAssembly, outputFile, expectedContent, responseContent);
+#else
+            Assert.Equal(expectedContent, responseContent, ignoreLineEndingDifferences: true);
+#endif
+        }
+
+        [Fact]
+        public async Task HomeController_Create_ReturnsSuccess()
+        {
+            // Arrange
+            var createBillyContent = CreateUserFormContent("Billy", "2000-11-30", 0, "hello");
+            var createJoeContent = CreateUserFormContent("Joe", "1980-10-20", 1, "goodbye");
+
+            // Act
+            var create = await Client.GetAsync("http://localhost/Home/Create");
+            var createBilly = await Client.PostAsync("http://localhost/Home/Create", createBillyContent);
+            var createJoe = await Client.PostAsync("http://localhost/Home/Create", createJoeContent);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, create.StatusCode);
+            AssertRedirectsToHome(createBilly);
+            AssertRedirectsToHome(createJoe);
+        }
+
+        [Fact]
+        public async Task HomeController_Edit_ReturnsSuccess()
+        {
+            // Arrange
+            var createBillyContent = CreateUserFormContent("Billy", "2000-11-30", 0, "hello");
+            var createJoeContent = CreateUserFormContent("Joe", "1980-10-20", 1, "goodbye");
+            var changeBillyContent = CreateUserFormContent("Bobby", "1999-11-30", 1, "howdy");
+            var changeBobbyContent = CreateUserFormContent("Bobby", "1999-11-30", 2, "howdy");
+            var changeJoeContent = CreateUserFormContent("Jack", "1979-10-20", 4, "goodbye");
+
+            // Act
+            var create = await Client.GetAsync("http://localhost/Home/Create");
+            var createBilly = await Client.PostAsync("http://localhost/Home/Create", createBillyContent);
+            var createJoe = await Client.PostAsync("http://localhost/Home/Create", createJoeContent);
+            var editBilly = await Client.GetAsync("http://localhost/Home/Edit/0");
+            var changeBilly = await Client.PostAsync("http://localhost/Home/Edit/0", changeBillyContent);
+            var changeBobby = await Client.PostAsync("http://localhost/Home/Edit/0", changeBobbyContent);
+            var changeJoe = await Client.PostAsync("http://localhost/Home/Edit/1", changeJoeContent);
```

Comment:
This is over-zealous, going well beyond the "more specific verification of pages" requested in #3530. Do you really need seven steps to show the `Edit()` action works? Only one existing functional test reaches `// Act 4` (actually `// Act 5` but someone repeated the number). Recommend 2 `[Fact]`s, one with two `.../Create` steps and another with one `../Create` and one `../Edit` step.